src/components/document-content.tsx

```diff
@@ -84,6 +96,37 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
     }
   }
 
+  private handleRowElementsChanged = (mutationsList: MutationRecord[], mutationsObserver: MutationObserver) => {
+    if (!this.domElement) return;
+
+    for (const mutation of mutationsList) {
+      if (mutation.type === "childList") {
+        // auto-scroll to new tile rows
+        if (mutation.addedNodes.length) {
+          const newRow = mutation.addedNodes[mutation.addedNodes.length - 1] as Element;
+          const newRowBounds = newRow.getBoundingClientRect();
+          const contentBounds = this.domElement.getBoundingClientRect();
+          const visibleContent = {
+                  top: this.domElement.scrollTop,
+                  bottom: this.domElement.scrollTop + contentBounds.height
+                };
+          const newRowInContent = {
+                  top: newRowBounds.top - contentBounds.top + this.domElement.scrollTop,
+                  bottom: newRowBounds.bottom - contentBounds.top + this.domElement.scrollTop
+                };
+          const kScrollTopMargin = 2;
+          const kScrollBottomMargin = 10;
+          if (newRowInContent.bottom > visibleContent.bottom) {
+            this.domElement.scrollTop += newRowInContent.bottom + kScrollBottomMargin - visibleContent.bottom;
+          }
+          else if (newRowInContent.top < visibleContent.top) {
+            this.domElement.scrollTop += newRowInContent.top - kScrollTopMargin - visibleContent.top;
+          }
```

Comment:
Why do you have to check both the top and the bottom of the new row?

Reply:
In case it was added offscreen above the viewable area, e.g. on drop.

Reply:
Good point, I forgot new rows aren't always appended